NEW Add phone link form for TinyMCE link option phone

Oliver996 opened the pull request:
<!--
  Thanks for contributing, you're awesome! ⭐

  Please read https://docs.silverstripe.org/en/contributing/code/ if you haven't contributed to this project recently.
-->
## Description
Adds the necessary forms for the phone link plugin in the TinyMCE. Is a part of this PR https://github.com/silverstripe/silverstripe-htmleditor-tinymce/pull/29

## Manual testing steps
Simply select the new the linking option in the TinyMCE "Link to phone number" and like the other linking option a popup opens where you can enter the number and text. After confirming, it creates a `<a href="tel:"></a>` with the text and phone number in it. Just like sslinkemail would do, only with a phone number.

## Issues
<!--
  List all issues here that this pull request fixes/resolves.
  If there is no issue already, create a new one! You must link your pull request to at least one issue.
-->
- https://github.com/silverstripe/silverstripe-htmleditor-tinymce/issues/28

## Pull request checklist
<!--
  PLEASE check each of these to ensure you have done everything you need to do!
  If there's something in this list you need help with, please ask so that we can assist you.
-->
- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [ ] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

Oliver996 committed:
NEW Add phone link form for TinyMCE link option phone

GuySartorelli reviewed (changes requested):
1. Why did you tick "CI is green" when CI hasn't run yet?
    - Please only check the boxes that are actually true. I have unticked it for now.
1. I have also unticked "Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release"
    - I couldn't find a corresponding changelog entry for CMS 6.2.0 in github.com/silverstripe/developer-docs/. Please create one.
1. Please remove the changes to files in the `lang/` directory. 
    - Localisations can be added through transifex later, as per [the contribution documentation](https://docs.silverstripe.org/en/6/contributing/translations/). 
    - You can keep the `en.yml` addition if you like (though it will be added there automatically next time we run our localisation script)

Oliver996 committed:
NEW Remove translations

Oliver996 commented:
Thank you for your review.

For the first point, I'm not fully sure if I did understand something wrong. When I added the commit, it did run the workflow. The first run two jobs were not green, but the error message did not really make sense for what I changed and when I restarted them they did run green. When I added the PR it was green https://github.com/biwac-websolutions-ag/silverstripe-admin/actions/runs/18717318097. Was that something else, or do I need to run the CI manually? There was one that was red, but that was my mistake because I did run it, not on the main branch. I remove that one because it's not needed. Only that I can do it correctly after this changes.

I did remove the translations. 

Then my last question is to the changelog. Can I add it to the CMS 6.1.0? Because I think it would be nice to already have it in this version, or is it too late for that, or did I add it in the wrong branch?

Sry for all the question it's my first time adding something to SilverStripe and not fully sure by all points. Thank you for your information.

GuySartorelli commented:
> For the first point, I'm not fully sure if I did understand something wrong. When I added the commit, it did run the workflow. The first run two jobs were not green, but the error message did not really make sense for what I changed and when I restarted them they did run green. When I added the PR it was green https://github.com/biwac-websolutions-ag/silverstripe-admin/actions/runs/18717318097.

Ahh. You were looking at CI in your forked repository - but the CI for this PR hadn't run, which is what we look at and rely on.
<img width="769" height="135" alt="image" src="https://github.com/user-attachments/assets/809173a9-b03f-48c7-85e3-4aaad5e71476" />

I'll let that run now since the requested change has been made.

> Then my last question is to the changelog. Can I add it to the CMS 6.1.0? Because I think it would be nice to already have it in this version, or is it too late for that, or did I add it in the wrong branch?

`6.1.0` was released earlier this month, so this will need to be included in the next minor release which is `6.2.0`.

> Sry for all the question it's my first time adding something to SilverStripe and not fully sure by all points. Thank you for your information.

No need to apologise, I'd much rather you ask questions and understand the process rather than just blindly acting. Thank you for your curiosity and making sure you're getting things right.

GuySartorelli reviewed (approved):
LGTM and works well locally, will merge now. Please let me know when you've got that changelog PR ready.

Oliver996 commented:
@GuySartorelli Thank you very much for your reply and explanation.
I have created the PR for the changelog https://github.com/silverstripe/developer-docs/pull/833
